platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/UnirestUtil.java

```diff
@@ -88,14 +114,33 @@ private static void validateRequest(String url, Map<String, String> headerParam)
 	}
 
 	private static Response getResponse(HttpResponse<String> response) {
+		Response resp = null;
 		if (null != response && StringUtils.isNotBlank(response.getBody())) {
 			try {
-				return mapper.readValue(response.getBody(), Response.class);
+				resp = mapper.readValue(response.getBody(), Response.class);
+				BACKOFF_DELAY = INITIAL_BACKOFF_DELAY;
 			} catch (Exception e) {
-				throw new ServerException("ERR_DATA_PARSER", "Unable to parse data! | Error is: " + e.getMessage());
+				LOGGER.error("UnirestUtil ::: getResponse ::: Error occurred while parsing api response. Error is: "+e.getMessage(), e);
+				if (BACKOFF_DELAY != INITIAL_BACKOFF_DELAY)
```

Comment:
I think the check should be : 
Instead of 
if (BACKOFF_DELAY != INITIAL_BACKOFF_DELAY)
it should be 
if (BACKOFF_DELAY != MAXIMUM_BACKOFF_DELAY)

Reply:
code changes done.